cmd/mailbox-controller/controller.go

```diff
@@ -253,51 +250,91 @@ func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 		logger.V(3).Info("Wanted workspace is being deleted, will retry later", "mbwsName", mbwsName)
 		return true
 	}
-	logger.V(3).Info("Both SyncTarget and Workspace exist and are not being deleted, now check on the APIBinding to edge", "mbwsName", mbwsName)
-	return ctl.ensureEdgeBinding(ctx, workspace)
+	logger.V(3).Info("Both SyncTarget and Workspace exist and are not being deleted, now check on the binding to edge", "mbwsName", mbwsName)
+	return ctl.ensureBinding(ctx, workspace)
 
 }
 
-const TheEdgeBindingName = "bind-edge"
-const TheEdgeExportName = "edge.kubestellar.io"
+func (ctl *mbCtl) ensureBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
+	logger := klog.FromContext(ctx).WithValues("mbsName", workspace.Name)
 
-func (ctl *mbCtl) ensureEdgeBinding(ctx context.Context, workspace *tenancyv1alpha1.Workspace) bool {
-	logger := klog.FromContext(ctx).WithValues("mbwsName", workspace.Name)
-	mbwsCluster := logicalcluster.Name(workspace.Spec.Cluster)
-	if mbwsCluster == "" {
-		logger.V(2).Info("Mailbox workspace does not have a Spec.Cluster yet")
-		return true
-	}
-	logger = logger.WithValues("mbwsCluster", mbwsCluster, "bindingName", TheEdgeBindingName)
-	scopedAPIBindingIfc := ctl.apiBindingClusterInterface.Cluster(mbwsCluster.Path())
-	theBinding, err := scopedAPIBindingIfc.Get(ctx, TheEdgeBindingName, metav1.GetOptions{})
-	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Failed to read APIBinding")
-		return true
-	}
-	if err == nil {
-		logger.V(4).Info("Found existing APIBinding, not checking spec", "spec", theBinding.Spec)
-		return false
-	}
-	binding := &apisv1alpha1.APIBinding{
-		ObjectMeta: metav1.ObjectMeta{
-			Name: TheEdgeBindingName,
-		},
-		Spec: apisv1alpha1.APIBindingSpec{
-			Reference: apisv1alpha1.BindingReference{
-				Export: &apisv1alpha1.ExportBindingReference{
-					Path: ctl.espwPath,
-					Name: TheEdgeExportName,
-				},
-			},
-		},
-	}
-	binding2, err := scopedAPIBindingIfc.Create(ctx, binding, metav1.CreateOptions{FieldManager: "TODO"})
-	if err != nil {
-		logger.Error(err, "Failed to create APIBinding", "binding", binding)
-		return true
+	// The script must be idempotent.
```

Comment:
Could we take the existence of the `APIServiceBinding` object to indicate that the binding has already been done? That seems like a win to me.

Reply:
The `kubestellar-kube-bind` script does check the CRD which indicates the binding has been done.

Reply:
done